Amazon.Ion.ObjectMapper/IonDictionarySerializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Amazon.IonDotnet;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Amazon.Ion.ObjectMapper
+{
+    public class IonDictionarySerializer : IonSerializer<IDictionary>
+    {
+        private IonSerializer serializer;
+        private Type valueType;
+
+        public IonDictionarySerializer(IonSerializer ionSerializer, Type valueType)
```

Comment:
Why do we need Type as a parameter, can we use IonSerializer<T> instead?

Reply:
We need the type because when we try to deserialize the value of the dictionary IonSerializer.Deserialize takes in a type parameter.

IonSerializer is not an IonSerializer<T>, which are internal serializers for different types. It is the overarching serializer that you use to use the object mapper which we need to recursively handle nested types.